src/NHapi.Base/SupportClass.cs

```diff
@@ -1,3268 +1,3486 @@
 /*

```

Comment:
im hoping much of this file is not needed but would take some investigation on where its being used.

Reply:
yeah - its enourmous - i've disabled warnings in this file and thought we could takle it separatley to this PR, what do you think?

Reply:
yup sounds good.

Reply:
will log a separate issue to refactor this `SupportClass`